charts/cryostat/templates/deployment.yaml

```diff
@@ -194,6 +194,35 @@ spec:
           securityContext:
             {{- toYaml .Values.storage.securityContext | nindent 12 }}
         {{- if not .Values.minimal }}
+        - name: {{ printf "%s-%s" .Chart.Name "oauth2proxy" }}
+          securityContext:
+            {{- toYaml .Values.oauth2Proxy.securityContext | nindent 12 }}
+          image: "{{ .Values.oauth2Proxy.image.repository }}:{{ .Values.oauth2Proxy.image.tag }}"
+          imagePullPolicy: {{ .Values.oauth2Proxy.image.pullPolicy }}
+          env:
+            - name: OAUTH2_PROXY_REDIRECT_URL
+              value: "http://localhost:8080/oauth2/callback"
+            - name: OAUTH2_PROXY_COOKIE_SECRET
+              value: {{ include "cryostat.cookieSecret" . }}
+            - name: OAUTH2_PROXY_PROVIDER
+              value: "google"
+            - name: OAUTH2_PROXY_CLIENT_ID
+              value: "CLIENT_ID"
+            - name: OAUTH2_PROXY_CLIENT_SECRET
+              value: "CLIENT_SECRET"
+            - name: OAUTH2_PROXY_EMAIL_DOMAINS
+              value: "*"
+            - name: UPSTREAM_CONFIG
+              value: {{ include "cryostat.base64Encode" "proxyRawPath: true" }}
```

Comment:
This doesn't look quite right... I think the alpha config must be passed as a path on the container start command, like this:

https://github.com/cryostatio/cryostat3/blob/328402ef8989f9aff9aec414cba265fdaae9f36f/smoketest/compose/auth_proxy.yml#L32

and the config should be a yaml file, not a base64-encoded string, like this one:

https://github.com/cryostatio/cryostat3/blob/main/smoketest/compose/auth_proxy_alpha_config.yaml

Reply:
I think we'll need to account for this alpha config somehow in downstream builds. It didn't look like the OpenShift oauth-proxy supports this.

Reply:
That's correct, unfortunately. We need it for oauth2-proxy because that's how you specify the behaviour to not re-encode path segments, and they only expose that config property through this YAML alpha config. On openshift-oauth-proxy the bugged re-encoding is always done, but it looks like with the upcoming Go version update it will not do it anymore and the bug is fixed.

I'm not sure how both projects will end up handling it since this is a behavioural change in the underlying Go standard library. oauth2-proxy doesn't use that anymore and instead uses a library dependency that allows them to have this configurable switch because they were afraid of changing behaviour and breaking compatibility. I don't know what openshift-oauth-proxy will end up doing or if they have the same concern (ie applications built to expect the proxy's bugged behaviour and work around it).

Reply:
My idea is that we will add some configuration switch for users to enable OpenShift integration. Turning that on is what will instruct the helm chart to deploy openshift-oauth-proxy instead of oauth2-proxy, so this will mean that not only the deployed image changes, but these specifics of how the containers are configured will change. That will also be an upstream feature, but deploying with openshift-oauth-proxy will always result in a broken deployment until they fix the path segment re-encoding bug anyway.